Import board from file

B-Mikembo opened the pull request.

Commit:
Factory creation for any word

Signed-off-by: Brice Mikembo <mikembo.brice@outlook.fr>

Commit:
import board from file

Signed-off-by: Brice Mikembo <mikembo.brice@outlook.fr>

Kwaaac reviewed (changes requested):
C'est super clean ce que tu as fait ! Je n'ai rien à redire à part deux trois truc pour chipoter ou des questions:

les levels, go les mettres dans les ressources !
    src/main/ressources/levels/...

- Dans EncryptionDecorator:
    -Méthode readData(): si je dis pas de connerie, si on utiliser un tableau de char au lieu d'un string, c'est avec la ligne 
            `buffer = new char[(int) file.length()];`
        Comme on sait la taille de notre lecture ?

    -Méthode initBoard(String data): est-ce qu'il ne serait pas plus malin en terme d'implémentation d'en faire une méthode 
        static de levelManager ? Comme c'est lui qui gère des boards. Voir, faire un type Board ? ça nous pemerttrait de garder 
        dans le type la taille du plateau ainsi que la liste des personnages jouable du plateau ? Comme ça dans le levelManager 
        on a juste un board 

     -Constructeur: je pense qu'il faudrait check si le fichier existe bien, comme dans le sujet il veulent pouvoir mettre le level 
     en ligne de commande, il vont forcément tester un nom au pif 
           ```java
           File tempFile = new File(levelDir + fileName);
           boolean exists = tempFile.exists();
           ```

      -Méthode widthBoard(String data): je pense que tu l'a corrigé de ton coté, mais il faut mettre un '\n' au lieu d'un '\r' sinon 
       j'ai une grosse erreur dans le main !, avec le '\n' tout se passe super bien


Bah franchement après avoir tout lu, c'est du lourd ce que tu as fait, gros gg !!! Les autres méthodes sont bonnes pour moi et j'ai beaucoup aimé le stream !!

Par contre les Factory, c'est une implémentation qui est excellente, à terme je pense que une factory d'Entity sera tout aussi smart et se melera bien avec decodeLine

Kwaaac committed:
add command line display (#2)

* Command line update

Draw the board in a the terminal

It is required to use a monospaced font

* Commande line Color update + display optimisation

* method name update

Commit:
WIP

Commit:
wip

Signed-off-by: Brice Mikembo <bmikembo@etud.u-pem.fr>

Kwaaac committed:
Vector & noun HasProperty update

Les vecteurs de coordonnées ont des ptis trucs sympas en plus tel que des factory pour les vecteur de déplacements + additions

les BordEntities, ont peut leurs demander si ils ont des propriétes, de base non(pour les noms, mais les entités (donc pas les noms) renvoie les propriétes quils ont selon leurs noms associé
